src/@ionic-native/plugins/hyper-track/index.ts

```diff
@@ -0,0 +1,246 @@
+
+  /**
+   * Complete an action from the SDK by its ID
+   *
+   * @param {String} [id] ID of the Action that will be marked as completed
+   *
+   * @returns {Promise<any>} Returns a Promise that resolves with an "OK" string if successful, or it gets rejected if an error ocurred.
+   *
+   * @see {@link https://github.com/hypertrack/hypertrack-cordova/blob/master/www/HyperTrack.js#39|HyperTrack.js completeAction implementation}
+   */
+  @Cordova()
+  completeAction(id: String): Promise<any> { return; }
+
+  /**
+   * Complete an action from the SDK using Action's lookupId as parameter
+   *
+   * @param {String} [lookupId] Lookup ID of the Action that will be marked as completed
+   *
+   * @returns {Promise<any>} Returns a Promise that resolves with an "OK" string if successful, or it gets rejected if an error ocurred.
+   *
+   * @see {@link https://github.com/hypertrack/hypertrack-cordova/blob/master/www/HyperTrack.js#46|HyperTrack.js completeActionWithLookupId implementation}
+   */
+  @Cordova()
+  completeActionWithLookupId(lookupId: String): Promise<any> { return; }
+
+  /**
+   * Disable the SDK and stop tracking.
+   * Needs user setting (via getOrCreateUser() or setUserId()) first.
+   *
+   * @returns {Promise<any>} Returns a Promise that resolves with the an "OK" string if successful, or it gets rejected if an error ocurred. One example of an error is not setting a User with getOrCreateUser() or setUserId() before calling this function.
+   *
+   * @see {@link https://github.com/hypertrack/hypertrack-cordova/blob/master/www/HyperTrack.js#52|HyperTrack.js stopTracking implementation}
+   */
+  @Cordova()
+  stopTracking(): Promise<any> { return; }
+
+  /**
+   * Get user's current location from the SDK
+   *
+   * @returns {Promise<any>} Returns a Promise that resolves with a string representation of the Location's JSON, or it gets rejected if an error ocurred.
+   *
+   * @see {@link https://github.com/hypertrack/hypertrack-cordova/blob/master/www/HyperTrack.js#58|HyperTrack.js getCurrentLocation implementation}
+   */
+  @Cordova()
+  getCurrentLocation(): Promise<any> { return; }
+
+  /**
+   * Check if Location permission has been granted to the app (for Android).
+   * Returns "true" or "false" in success method accordingly.
+   *
+   * @returns {Promise<any>} Returns a Promise that resolves with the a string that can be "true" or "false", depending if location permission was granted, or it gets rejected if an error ocurred.
+   *
+   * @see {@link https://github.com/hypertrack/hypertrack-cordova/blob/master/www/HyperTrack.js#64|HyperTrack.js checkLocationPermission implementation}
+   * @see {@link https://docs.hypertrack.com/sdks/android/setup.html#step-3-enable-location-permissions|HyperTrack Android SDK Setup}
+   */
+  @Cordova({
+    platforms: ['android']
```

Comment:
I have been in touch with the maintainer of the HyperTrack cordova plugin, and iOS support is coming soon, however the methods that I specified that they only work on Android will most probably only be available for Android, even when iOS support comes. This is: `checkLocationPermission`, `requestPermissions`, `checkLocationServices`, and `requestLocationServices `will only work in Android even when iOS is supported.

This is the reason why I specified it, but I can erase it, and once iOS is supported it can be put back.